src/main/java/io/snyk/jenkins/SnykStepBuilder.java

```diff
@@ -380,39 +368,40 @@ ArgumentListBuilder buildArgumentList(String snykExecutable, String snykCommand,
     return args;
   }
 
-  private void generateSnykHtmlReport(Run<?, ?> build, @Nonnull FilePath workspace, Launcher launcher, TaskListener log, String reportExecutable, String monitorUri) throws IOException, InterruptedException {
-    EnvVars env = build.getEnvironment(log);
-    ArgumentListBuilder args = new ArgumentListBuilder();
-
-    FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
-    if (!snykReportJson.exists()) {
-      log.getLogger().println("Snyk report json doesn't exist");
-      return;
-    }
-
-    workspace.child(SNYK_REPORT_HTML).write("", UTF_8.name());
-
-    args.add(reportExecutable);
-    args.add("-i", SNYK_TEST_REPORT_JSON, "-o", SNYK_REPORT_HTML);
-    try {
-      int exitCode = launcher.launch()
-                             .cmds(args)
-                             .envs(env)
-                             .quiet(true)
-                             .pwd(workspace)
-                             .join();
-      boolean success = exitCode == 0;
-      if (!success) {
-        log.getLogger().println("Generating Snyk html report was not successful");
-      }
-      String reportWithInlineCSS = workspace.child(SNYK_REPORT_HTML).readToString();
-      String modifiedHtmlReport = ReportConverter.getInstance().modifyHeadSection(reportWithInlineCSS, Jenkins.get().servletContext.getContextPath());
-      String finalHtmlReport = ReportConverter.getInstance().injectMonitorLink(modifiedHtmlReport, monitorUri);
-      workspace.child(workspace.getName() + "_" + SNYK_REPORT_HTML).write(finalHtmlReport, UTF_8.name());
-    } catch (IOException ex) {
-      Util.displayIOException(ex, log);
-      ex.printStackTrace(log.fatalError("Snyk-to-Html command execution failed"));
+  private String generateSnykHtmlReport(
+    Run<?, ?> build,
+    @Nonnull FilePath workspace,
+    Launcher launcher,
+    TaskListener listener,
+    String reportExecutable,
+    String testJson,
+    String monitorUri
+  ) throws IOException, InterruptedException {
+    EnvVars env = build.getEnvironment(listener);
+    PrintStream logger = listener.getLogger();
+    ArgumentListBuilder args = new ArgumentListBuilder()
+      .add(reportExecutable);
+    ByteArrayOutputStream htmlStdout = new ByteArrayOutputStream();
+
+    int exitCode = launcher.launch()
+      .cmds(args)
+      .envs(env)
+      .stdin(new ByteArrayInputStream(testJson.getBytes(UTF_8.name())))
```

Comment:
Keep the byte stream and convert to an input stream instead of converting it to a string and then a input stream.